### DISCUSSION TITLE
Supporting Illustrator layers

### DISCUSSION THREAD

**Discussion by Andreasox** [pymupdf/PyMuPDF#3054] [Q&A]
PDF layers AKA OCGs created with PyMuPDF works good.
However, Illustrator does not recognize PDF layers (to our and many others major surprise).
Illustrator seems to store its internal layer info in PDFs exported from Illustrator according to the answer given in https://community.apryse.com/t/editing-layers-in-adobe-illustrator-pdf-s-using-pdfnet-sdk/985:
"Illustrator stores its private data under 'PieceInfo/Illustrator/Private' in the page dictionary".
Is it possible to enhance PyMuPDF to also support Illustrator layers in the same way as OCG?

**Comment by JorjMcKie** [comment]
Very sorry to give you another "No" answer ... but "No" 🤷‍♂️.

**Reply by Andreasox** [reply]
I would have been really surprised for any other answer. May I ask if you have investigated the complexities of this? 
My guess is that T-Kartor will give it a try, after reverse-engineering PDFs created by Illustrator.

**Comment by wedmonds** [comment]
Any updates on this? I'm working on a project that takes a PDF (produced by Illustrator with certain Ink properties per layer) and am trying to alter color and graphics programmatically on a per layer basis.

**Reply by JorjMcKie** [reply]
@wedmonds - No news here.
We also do not intend to pursue this in the foreseeable future.

**Comment by Andreasox** [comment]
We ended by creating one PDF with PyMuPDF for each desired layer, and programmatically combined them in the Adobe environment.
